test/gtest/uct/test_md.cc

```diff
@@ -492,6 +492,56 @@ UCS_TEST_SKIP_COND_P(test_md, sockaddr_accessibility,
     freeifaddrs(ifaddr);
 }
 
+UCS_TEST_SKIP_COND_P(test_md, fork,
+                     !check_caps(UCT_MD_FLAG_REG),
+                     "RCACHE_CHECK_PFN=1")
+{
+    uct_md_attr_t md_attr;
+    ucs_status_t status;
+    int thread_status;
+    uct_mem_h memh;
+    char *page;
+    pid_t pid;
+
+    status = uct_md_query(md(), &md_attr);
+    ASSERT_UCS_OK(status);
+
+    if (!(md_attr.cap.reg_mem_types & UCS_BIT(UCS_MEMORY_TYPE_HOST))) {
+        UCS_TEST_SKIP_R("not host memory type");
+    }
+
+    page = (char *)valloc(100);
+    memset(page, 42, 100);
+
+    status = uct_md_mem_reg(md(), page, 100, UCT_MD_MEM_ACCESS_ALL, &memh);
+    ASSERT_UCS_OK(status);
+    ASSERT_TRUE(memh != UCT_MEM_HANDLE_NULL);
+
+    status = uct_md_mem_dereg(md(), memh);
+    EXPECT_UCS_OK(status);
+
+    pid = fork();
+    if (pid == 0) {
+        EXPECT_EQ(0, memcmp(page, page + 50, 50));
+        exit(0);
+    }
+
+    EXPECT_NE(-1, pid);
+    memset(page, 42, 100);
+
+    status = uct_md_mem_reg(md(), page, 100, UCT_MD_MEM_ACCESS_ALL, &memh);
+    ASSERT_UCS_OK(status);
+    ASSERT_TRUE(memh != UCT_MEM_HANDLE_NULL);
+
+    status = uct_md_mem_dereg(md(), memh);
+    EXPECT_UCS_OK(status);
```

Comment:
why would it fail without the fork fix?
do we need to make sure this happens before child touches the page? If yes child needs to sleep before / use pipe to sync

Reply:
without fork fix knem would fail pfn check on mem_reg because of rcache decogeration

Reply:
so the child needs to exit before mem_reg is done on parent? need to move wait before mem_reg?

Reply:
no, invalidation triggered by do_fork()

Reply:
but if the rcache was not cleared, we would need to wait for child to finish to have PFN check fail, right?